### PR TITLE
Refactored setvcpus testcase

### PR DESCRIPTION
**Pull request by sathnaga**
1. Added expected output check in all related virsh APIs.
2. Removed autotest dependancy.
3. Modified maximum vcpu change based on a upstream bug,
details in the code comment.
4. few more minor fixes.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sathnaga** [comment]
This PR has this dependency https://github.com/avocado-framework/avocado-vt/pull/939

**Comment by sathnaga** [comment]
```
 (01/36) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (52.03 s)
 (02/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_live.add: PASS (41.95 s)
 (03/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_live.del: PASS (43.87 s)
 (04/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_config.add: PASS (42.19 s)
 (05/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_config.del: PASS (41.98 s)
 (06/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_config_live.add: PASS (52.18 s)
 (07/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_config_live.del: PASS (44.75 s)
 (08/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_current.add: PASS (40.82 s)
 (09/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.id_option.option_current.del: PASS (41.45 s)
 (10/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.add: PASS (40.55 s)
 (11/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.del: PASS (40.35 s)
 (12/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.pause_option.add: PASS (40.79 s)
 (13/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.pause_option.del: PASS (38.25 s)
 (14/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.uuid_option.add: PASS (41.03 s)
 (15/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_on.uuid_option.del: PASS (40.80 s)
 (16/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_off.option_config: PASS (46.06 s)
 (17/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_off.option_uuid_config: PASS (41.49 s)
 (18/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_off.option_current: PASS (46.77 s)
 (19/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.guest_off.option_maximum_config: PASS (47.24 s)
 (20/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.normal_test.remote: CANCEL (30.42 s)
 (21/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.no_option: PASS (41.15 s)
 (22/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.unexcept_id_option.hex_id_option: PASS (41.89 s)
 (23/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.unexcept_id_option.id_current_live_option: PASS (41.32 s)
 (24/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.unexcept_id_option.invalid_id_option: PASS (42.61 s)
 (25/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.unexpected_domain_option: PASS (40.50 s)
 (26/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.invalid_uuid_option: PASS (42.01 s)
 (27/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.extra_option: PASS (41.34 s)
 (28/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.name_only_option: PASS (40.83 s)
 (29/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.invalid_vcpu_count_is_0: PASS (41.58 s)
 (30/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.invalid_vcpu_count_max: PASS (41.32 s)
 (31/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.shut_off_error_option.live_option: PASS (46.61 s)
 (32/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.shut_off_error_option.live_config_option: PASS (44.77 s)
 (33/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.shut_off_error_option.current_config_option: PASS (46.04 s)
 (34/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.shut_off_error_option.with_topology: PASS (38.51 s)
 (35/36) guest_cpu.setvcpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.virsh.setvcpus.error_test.maximum_option: PASS (42.58 s)
 (36/36) guest_remove.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.25.ppc64le.powerkvm-libvirt.remove_guest.without_disk: PASS (2.70 s)
RESULTS    : PASS 35 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
TESTS TIME : 1490.76 s

```

**Comment by sathnaga** [comment]
@will-Do Thanks, could you please approve the dependency PR if you agree to the place and no other further comments, let me know otherwise, Thanks in advance :-)

**Comment by will-Do** [comment]
@sathnaga Sure, will take a look that PR later

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-
 from virttest import remote
 from virttest import virsh
 from virttest import libvirt_vm
 from virttest.libvirt_xml import vm_xml
-from virttest.utils_test import libvirt
+from virttest import utils_hotplug
 
 
 def remote_test(remote_ip, local_ip, remote_pwd, remote_prompt,
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: As part of refactor, we can use `virsh.setvcpus()` with `uri` and `check_exit_status` for remote test case can remove this function.

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py [hunk 2]
```diff
@@ -29,8 +25,8 @@ def remote_test(remote_ip, local_ip, remote_pwd, remote_prompt,
         session.cmd_output('LANG=C')
         command = "virsh -c %s setvcpus %s 1 --live" % (remote_uri, vm_name)
         if virsh.has_command_help_match("setvcpus", "--live") is None:
-            raise error.TestNAError("The current libvirt doesn't support"
-                                    " '--live' option for setvcpus")
+            test.cancel("The current libvirt doesn't support"
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: skipping the tests before starting the test will be more appropriate

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py (hunk 1 above)
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: @balamuruhans Sure will address it with a separate PR, needs few tests around it.

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py (hunk 2 above)
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: @balamuruhans done

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py [hunk 3]
```diff
@@ -99,42 +54,73 @@ def run(test, params, env):
...
-        current_vcpu = int(current_vcpu)
+        max_vcpu = int(params.get("setvcpus_max", "4"))
     except ValueError:
-        raise error.TestError(convert_err.format(current_vcpu))
-    max_vcpu = int(params.get("setvcpus_max", "4"))
+        test.error(convert_err.format(max_vcpu))
     try:
-        max_vcpu = int(max_vcpu)
+        count = params.get("setvcpus_count", "")
+        if count != "":
```
Comment (sathnaga) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: @balamuruhans done :-)

**Comment by balamuruhans** [comment]
Thanks @sathnaga @will-Do

**Pull request merged**